src/main/kotlin/org/rust/ide/intentions/DemorgansLawIntention.kt

```diff
@@ -0,0 +1,107 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.elementType
+import org.rust.lang.core.psi.util.parentOfType
+
+class DemorgansLawIntention : PsiElementBaseIntentionAction() {
+    override fun getText() = "DeMorgan Law"
```

Comment:
@mkaput and i think there is no need to introduce a base class here, i mean to dynamic change the intention text by adding a base class is a little bit overstate.